File: common/src/main/scripts/lib/micropython/oc2/events.py
from oc2.channel import Channel


class Events:
    def __init__(self, channel):
        self.channel = channel

    @classmethod
    def from_stream(cls, stream, max_frame=None):
        if max_frame is None:
            return cls(Channel(stream))
        return cls(Channel(stream, max_frame))

    @classmethod
    def open(cls, path):
        return cls(Channel.open(path, read_only=True))

    def close(self):
        self.channel.close()

    def poll(self):
        return self.channel.read(0)

    def wait(self, timeout=None):
        return self.channel.read(timeout)

File: common/src/main/scripts/lib/lua/oc2/events.lua
local Channel = require("oc2.channel")

local Events = {}
Events.__index = Events

function Events.fromChannel(channel)
  return setmetatable({ channel = channel }, Events)
end

function Events.fromFd(fd, maxFrame)
  return Events.fromChannel(Channel.fromFd(fd, maxFrame))
end

function Events.open(path)
  local channel, reason = Channel.open(path, true)
  if not channel then
    return nil, reason
  end
  return Events.fromChannel(channel)
end

function Events:close()
  self.channel:close()
end

function Events:poll()
  return self.channel:read(0)
end

function Events:wait(timeout)
  return self.channel:read(timeout)
end

return Events
